udp.go

```diff
@@ -37,23 +37,36 @@ func (s *SessionUDP) RemoteAddr() net.Addr { return s.raddr }
 
 // ReadFromSessionUDP acts just like net.UDPConn.ReadFrom(), but returns a session object instead of a
 // net.UDPAddr.
-func ReadFromSessionUDP(conn *net.UDPConn, b []byte) (int, *SessionUDP, error) {
-	oob := make([]byte, udpOOBSize)
-	n, oobn, _, raddr, err := conn.ReadMsgUDP(b, oob)
+func ReadFromSessionUDP(conn net.PacketConn, b []byte) (int, *SessionUDP, error) {
+	if netUDPConn, ok := conn.(*net.UDPConn); ok {
```

Comment:
how does this type switch affect the throughput?

Reply:
Good question! how can I prove it ? 
I guess it depends on the number of incoming requests. A load test might work. I take a look at dnsperf.

Reply:
I tried a simple udp server and performance are quite similar. https://gist.github.com/guillaumerose/7b59481a9a56eec9f4c2c7918dc0333f